src/components/filters/filter_icons_overlay/fitler_icons_overlay.ts

```diff
@@ -32,11 +32,19 @@ export class FilterIconsOverlay extends Component<Props, SpreadsheetChildEnv> {
     const rowDims = this.env.model.getters.getRowDimensionsInViewport(sheetId, position.row);
     const colDims = this.env.model.getters.getColDimensionsInViewport(sheetId, position.col);
 
-    // TODO : change this offset when we support vertical cell align
     const centeringOffset = Math.floor((rowDims.size - FILTER_ICON_EDGE_LENGTH) / 2);
+    const cell = this.env.model.getters.getCell({ sheetId, col: position.col, row: position.row });
+    let verticalFilterIconPosition =
+      rowDims.end - FILTER_ICON_EDGE_LENGTH + this.props.gridPosition.y - centeringOffset;
+    if (cell?.style?.verticalAlign === "bottom") {
+      verticalFilterIconPosition =
+        rowDims.end - FILTER_ICON_EDGE_LENGTH + this.props.gridPosition.y - FILTER_ICON_MARGIN;
+    } else if (cell?.style?.verticalAlign === "top") {
+      verticalFilterIconPosition = rowDims.start + this.props.gridPosition.y + FILTER_ICON_MARGIN;
+    }
```

Comment:
The computations look correct, but it's a bit hard to read. To increase readability, maybe you should : 

- extract all that to a function `private getIconVerticalPosition()`
- replace the if/else by a `switch(cell.verticalAlign)`
- add `this.props.gridPosition.y` at the end after the computations, sicne we do it every time (maybe directly in the `return` of `getFilterHeaderPosition` ?)